it/controllers/ChargeSummaryControllerISpec.scala

```diff
@@ -412,6 +412,240 @@ class ChargeSummaryControllerISpec extends ComponentSpecBase {
+            outstandingAmount = Some(1200),
+            documentDate = LocalDate.of(2018, 3, 29)
+          ),
+          dueDate = Some(LocalDate.parse("2018-03-30"))
+        ),
+        paymentBreakdown = List(),
+        chargeHistories = List.empty,
+        paymentAllocations = List.empty,
+        isLatePaymentCharge = false,
+        isMFADebit = true
+      ))
+
+    }
+
+    "load the charge summary page with a PAID MFADebit" in {
+      Given("the MFADebitsAndCredits feature switch is enabled")
+      enable(MFACreditsAndDebits)
+      enable(ChargeHistory)
+      enable(PaymentAllocation)
+
+      Given("I wiremock stub a successful Income Source Details response with property only")
+      IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndPropertyResponse)
+
+      And("I wiremock stub a single financial transaction response")
+      IncomeTaxViewChangeStub.stubGetFinancialDetailsByDateRange(testNino)(OK, financialDetailsPaidMFA)
+
+      And("I wiremock stub a charge history response")
+      IncomeTaxViewChangeStub.stubChargeHistoryResponse(testMtditid, "1040000124")(OK, testChargeHistoryJson(testMtditid, "1040000124", 2500))
+
+      val res = IncomeTaxViewChangeFrontend.getChargeSummary(s"$testTaxYear", "1")
+
+      verifyIncomeSourceDetailsCall(testMtditid)
+
+      val summaryListText = "Due date 30 March 2018 Full payment amount £1,200.00 Remaining to pay £0.00"
+      val hmrcCreated = messagesAPI("chargeSummary.chargeHistory.created.hmrcAdjustment.text")
+      val paymentHistoryText = "Date Description Amount 29 Mar 2018 " + hmrcCreated + " £1,200.00"
+      val paymentHistoryText2 = "28 Jul 2022 Payment put towards HMRC adjustment 2018 £1,200.00"
+
+      Then("the result should have a HTTP status of OK (200) and load the correct page")
+      res should have(
+        httpStatus(OK),
+        pageTitleIndividual("chargeSummary.hmrcAdjustment.text"),
+        elementTextBySelector(".govuk-summary-list")(summaryListText),
+        elementCountBySelector(s"#payment-link-$testTaxYear")(0),
+        elementCountBySelector("#payment-history-table tr")(3),
+        elementTextBySelector("#payment-history-table tr:nth-child(1)")(paymentHistoryText),
+        elementTextBySelector("#payment-history-table tr:nth-child(2)")(paymentHistoryText2)
+      )
+
+      AuditStub.verifyAuditEvent(ChargeSummaryAudit(
+        MtdItUser(
+          testMtditid, testNino, None,
+          multipleBusinessesAndPropertyResponse, None, Some("1234567890"),
+          Some("12345-credId"), Some("Individual"), None
+        )(FakeRequest()),
+        DocumentDetailWithDueDate(
+          documentDetail = DocumentDetail(
+            taxYear = "2018",
+            transactionId = "1040000124",
+            documentDescription = Some("TRM New Charge"),
+            documentText = Some("documentText"),
+            originalAmount = Some(1200),
+            outstandingAmount = Some(0),
```

Comment:
This figure is the only one that changes? Could pass it in instead?

Reply:
sure will do